src/events.js

```diff
@@ -95,22 +95,22 @@ function handleClickEvents(chart, state, event, options) {
       // 2nd click before timeout, so its a double click
       clearTimeout(element.clickTimeout);
       delete element.clickTimeout;
-      dispatchEvent(chart, dblclick, element, event);
+      dispatchEvent(dblclick, element, event);
     } else if (dblclick) {
       // if there is a dblclick handler, wait for dblClickSpeed ms before deciding its a click
       element.clickTimeout = setTimeout(() => {
         delete element.clickTimeout;
-        dispatchEvent(chart, click, element, event);
+        dispatchEvent(click, element, event);
       }, options.dblClickSpeed);
     } else {
       // no double click handler, just call the click handler directly
-      dispatchEvent(chart, click, element, event);
+      dispatchEvent(click, element, event);
     }
   }
 }
 
-function dispatchEvent(chart, handler, element, event) {
-  callHandler(handler, [{chart, element}, event]);
+function dispatchEvent(handler, element, event) {
+  callHandler(handler, [element.$context, event]);
```

Comment:
Does the element **always** contain `$context`?
I'd think it does not when there is nothing scriptable, but that maybe not the case?

Reply:
Answering to myself, its always there. And it set there (in `updateElements`):

https://github.com/chartjs/chartjs-plugin-annotation/blob/405433949306d949e4460e8e65487b4baa650346/src/annotation.js#L142